src/GitVersionCore/Git/ReferenceName.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+
+namespace GitVersion
+{
+    public class ReferenceName
```

Comment:
I think this class turned out great! Perhaps we should add a `ToString()` method returning `FriendlyName` and have it implement `IEquatable<ReferenceName>` and `IComparable<ReferenceName>` as well?

Reply:
Yeap, I will do that

Reply:
Done